mantatail.py

```diff
@@ -56,28 +87,49 @@ def run_server_forever(self) -> None:
 
 
 def close_socket_cleanly(sock: socket.socket) -> None:
+    """
+    Ensures that the connection to a client is closed cleanly without errors and with no data loss.
+
+    Use this instead of the .close() method.
+    """
+    # The code is based on this blog post:
     # https://blog.netherlabs.nl/articles/2009/01/18/the-ultimate-so_linger-page-or-why-is-my-tcp-not-reliable
     try:
         sock.shutdown(socket.SHUT_WR)
         sock.settimeout(10)
         sock.recv(1)  # Wait for client to close the connection
     except OSError:
         # Possible causes:
-        #   - Client decided to keep its connection open for more than 10sec.
-        #   - Client was already disconnected.
-        #   - Probably something else too that I didn't think of...
+        # - Client decided to keep its connection open for more than 10sec.
+        # - Client was already disconnected.
+        # - Probably something else too that I didn't think of...
         pass
 
     sock.close()
 
 
 def recv_loop(state: ServerState, user_host: str, user_socket: socket.socket) -> None:
+    """
+    Instantiates UserConnection and listens for commands/messages from the client,
```

Comment:
- Instantiating UserConnection is an implementation detail. It doesn't belong in the docstring.
- This is not listening; that means accepting new client connections. It is receiving

```suggestion
    Receives commands/messages from the client,
```